src/txn/txn.c

```diff
@@ -1211,6 +1244,19 @@ __txn_resolve_prepared_op(WT_SESSION_IMPL *session, WT_TXN_OP *op, bool commit,
     cbt = (WT_CURSOR_BTREE *)(*cursorp);
     page = cbt->ref->page;
 
+    /*
+     * For in-memory configurations of WiredTiger if a prepared update is reconciled and then rolled
+     * back the on-page value will not be marked as aborted until the next eviction. In the special
+     * case where this rollback results in the update chain being entirely comprised of aborted
+     * updates other transactions attempting to write to the same key will look at the on-page
+     * value, think the prepared transaction is still active, and falsely report a write conflict.
+     * To prevent this scenario append a tombstone to the update chain when rolling back a prepared
+     * update would result in only aborted updates on the update chain.
+     */
+    if (F_ISSET(S2C(session), WT_CONN_IN_MEMORY) &&
+      F_ISSET(upd, WT_UPDATE_PREPARE_RESTORED_FROM_DS) && !commit && first_committed_upd == NULL)
+        WT_ERR(__txn_append_tombstone(session, op, cbt));
```

Comment:
I feel the logic can be simplified if we combine these in if else if structure instead of using separate ifs

Reply:
Fixed in fe83473